src/grt/grt-wave_options-parse.adb

```diff
@@ -0,0 +1,323 @@
+         if Current_Version.Minor /= 0 then
+            Error_C ("between ");
+            Print_Version (Version_Type'(Current_Version.Major, 0));
+            Error_C (" and ");
+         end if;
+         Print_Version (Current_Version);
+         Error_E;
+      end if;
+
+   end Parse_Version;
+
+   procedure Print_Version (Version : Version_Type) is
+      Num_Str : String (1 .. Value_String_Size);
+      First : Positive;
+   begin
+      To_String (Num_Str, First, Ghdl_I32 (Version.Major));
+      Report_C (Num_Str (First .. Num_Str'Last));
+      Report_C (".");
+      To_String (Num_Str, First, Ghdl_I32 (Version.Minor));
+      Report_C (Num_Str (First .. Num_Str'Last));
+   end Print_Version;
+
+   --------------------------------------------------------------------------
+
+   procedure Parse_Path (Line : String_Acc) is
+      First, Last : Positive;
+      Tree_Updated : Boolean;
+      Tree_Index : Tree_Index_Type;
+   begin
+      To_Lower (Line_Context.Str.all);
+      Last := Line'First;
+      if Line (Line'First) = '/' then
+         Tree_Index := Entity;
+         Last := Last + 1;
+      else
+         Tree_Index := Pkg;
+      end if;
+      Tree_Cursor := Trees (Tree_Index);
+      Previous_Tree_Cursor := null;
+
+      loop
+         First := Last;
+
+         loop -- Find next identifier
+            if Line (Last) = Sep (Tree_Index) then
+               Last := Last - 1;
+               exit;
+            elsif Last = Line'Last then
+               exit;
+            end if;
+            Last := Last + 1;
+         end loop;
+
+         Check_Validity (Line (First .. Last));
+         Tree_Updated := Update_Tree (Line (First .. Last), Tree_Index);
+         Line_Context.Max_Level := Line_Context.Max_Level + 1;
+
+         if Last = Line'Last then
+            if not Tree_Updated then
+               Error_Context ("ignored already known signal path", Warning);
+            end if;
+            return;
+         end if;
+
+         Last := Last + 2; -- Skip the separator
+         if Last > Line'Last then -- catch signal paths ending with /
+            Error_Context ("invalid signal path");
+         end if;
+
+      end loop;
+
+   end Parse_Path;
+
+   function Update_Tree (Elem_Name : String; Tree_Index : Tree_Index_Type)
+            return Boolean is
+      Sibling_Cursor : Elem_Acc := Tree_Cursor;
+      Previous_Sibling_Cursor : Elem_Acc := null;
+      Elem : Elem_Acc;
+   begin
+      loop
+         -- Already reached the last sibling and current identifier corresponds
+         -- to no existing element ? Then we will create an element
+         if Sibling_Cursor = null then
+            Elem := new Elem_Type'(Name => new String'(Elem_Name),
+                                   Line_Context => Line_Context,
+                                   Kind => Not_Found,
+                                   Next_Sibling | Next_Child => null);
+            if Previous_Sibling_Cursor = null then -- First element of level ?
```

Comment:
Sorry I forgot to do the changes everywhere when you first pointed the 
problem. Now it is Fixed.